src/include/migraphx/shape.hpp

```diff
@@ -107,6 +107,11 @@ struct shape
         friend bool operator==(const std::size_t& x, const dynamic_dimension& y);
         friend bool operator!=(const dynamic_dimension& x, const std::size_t& y);
         friend bool operator!=(const std::size_t& x, const dynamic_dimension& y);
+
+        // add and subtract fixed std::size_t dimension
+        friend dynamic_dimension operator+(const dynamic_dimension& x, const std::size_t& y);
+        friend dynamic_dimension operator+(const std::size_t& x, const dynamic_dimension& y);
+        friend dynamic_dimension operator-(const dynamic_dimension& x, const std::size_t& y);
```

Comment:
No subtract dynamic_dimension from size_t?  Are we assuming this result is just a size_t then and handled somewhere else?

Reply:
I don't use that case in the current code. I will add it and some asserts for the subtraction operators to catch if the subtraction goes below 0. (a dimension of 0 won't make sense for most cases either?)

Reply:
It's a somewhat weird case however.
Example:
10 - {2, 5, 3} = {5, 8, 7} where the `min` output is `10 - max` and `max` output is `10 - min`

Reply:
Holding off on adding that case. I don't see where we would use it.

Reply:
Thats fine. Just wanted to ask the question since you had the other cases with the addition case.